scripts/packaging/versionmetadata.ps1

```diff
@@ -0,0 +1,238 @@
+        }
+    }
+}
+
+<#
+.SYNOPSIS
+    Adds AssemblyInfo.cs files to all locations within the Assets/ folder that
+    have an .asmdef file.
+
+    If AssemblyInfo.cs already exists, this will instead just add the version
+    information to the file.
+
+    Returns true if an issue is found.
+#>
+function AddAssemblyInfo {
+    [CmdletBinding()]
+    param(
+        [string]$Directory,
+        [string]$Version
+    )
+    process {
+        $containsIssue = $false
+        $assets = Join-Path -Path $Directory -ChildPath "Assets"
+        $mrtkFolder = Join-Path -Path $assets -ChildPath "MRTK"
+        $asmdefs = Get-ChildItem $assets *.asmdef -Recurse | Select-Object FullName
+        foreach ($asmdef in $asmdefs) {
+            # The AssemblyInfo.cs file will be added as a sibling of the .asmdef location,
+            # so we need to trim off the filename.
+            $folder = Split-Path -Path $asmdef.FullName
+            $filename = Join-Path -Path $folder -ChildPath "AssemblyInfo.cs"
+
+            # We also need to parse out the MRTK subfolder (i.e. Core, SDK, Examples)
+            # which goes into the AssemblyProduct metadata.
+            $projectFolder = $filename.Replace($mrtkFolder, "")
+            $split = $projectFolder.Split([IO.Path]::DirectorySeparatorChar)
+
+            # Since $projectFolder has the form "/SDK/Path/Thing"
+            # we use index 1 to get the value "SDK"
+            # Also note that there's a space ahead of the value which is intentional
+            # to ensure that resolved product looks like:
+            # "Microsoft® Mixed Reality Toolkit SDK" when in a non-Core folder
+            # and "Microsoft® Mixed Reality Toolkit" when in the Core folder.
+            $project = " " + $split[1]
+            if ($project -eq " Core") {
+                $project = ""
+            }
+
+            # Note that this is left-indent adjusted so that the file output
+            # ends up looking reasonable.
+            $copyright = 
+@"
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+"@
+
+            $content =
+@"
+[assembly: System.Reflection.AssemblyVersion("$Version.0")]
+[assembly: System.Reflection.AssemblyFileVersion("$Version.0")]
+
+[assembly: System.Reflection.AssemblyProduct("Microsoft® Mixed Reality Toolkit$project")]
+[assembly: System.Reflection.AssemblyCopyright("Copyright © Microsoft Corporation")]
+"@
+
+            # If the AssemblyInfo already exists, then we should check that it doesn't already contain
+            # a version - this would be bad as we'd duply-define the version
+            if (Test-Path -Path $filename) {
+                if (Get-Content $filename | Select-String "AssemblyVersion") {
+                    $containsIssue = $true
+                    Write-Host "$filename should not have an AssemblyVersion declaration, this is added at CI time"
```

Comment:
Do I understand correctly that the version update is only ever going to be run on CI, so we don't need to worry about re-running it in a local repo ?

Reply:
Yep - the goal here was to make it so that we don't have any per-file version numbers to avoid getting things out of sync there. If you are building binaries yourself (i.e. offline) you'll still need to run this step however, though it's not something the team is usually doing